### PR TITLE
Fixed switch role check for IDF isolation configuration

### PR DESCRIPTION
**Pull request by tjchadaga**
#### Why I did it
Fixes https://github.com/sonic-net/sonic-buildimage/issues/19379
 
##### Work item tracking
- Microsoft ADO **(number only)**: 28691372

#### How I did it
Changed the lookup from switch_type (which is specific to chassis) to type (which holds the role for all pizzabox and chassis)

#### How to verify it
sudo idf_isolation <> commands run on chassis shoud add route-maps. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by gechiang — APPROVED** [review]
LGTM

**Comment by tjchadaga** [comment]
@lguohan, @StormLiangMS - could you please help merge this?

File: src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py [hunk 1]
```diff
@@ -259,8 +257,8 @@ def downstream_isolate_unisolate(self, idf_isolation_state):
             log_err("IDF: invalid value({}) is provided".format(idf_isolation_state))
             return False
 
-        if self.switch_type and self.switch_type != "SpineRouter":
-            log_debug("DeviceGlobalCfgMgr:: Skipping IDF isolation configuration on Switch type: %s" % self.switch_type)
+        if self.switch_role and self.switch_role != "SpineRouter":
+            log_debug("DeviceGlobalCfgMgr:: Skipping IDF isolation configuration on %s" % self.switch_role)
```
Comment (kperumalbfn) [inline src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py:261]: Do we need this change in 202311, 202405?

Comment (tjchadaga) [reply src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py:261]: needed for 202405, not 202311

**Review by kperumalbfn — APPROVED** [review]
LGTM

**Comment by gechiang** [comment]
@tjchadaga do we need this for 202205 MSFT repo so this is also available to the community using the MSFT repo 202205 branch for chassis project?  If so, please tag it.

**Comment by bingwang-ms** [comment]
@tjchadaga If the PR is required in 202405 branch, please file a direct PR to 202405 to address cherry-pick conflict.

**Comment by tjchadaga** [comment]
> @tjchadaga do we need this for 202205 MSFT repo so this is also available to the community using the MSFT repo 202205 branch for chassis project? If so, please tag it.

not required in msft repo 202205, since the original PR that introduced the change (https://github.com/sonic-net/sonic-buildimage/pull/19136) is not in that repo

**Comment by tjchadaga** [comment]
> @tjchadaga If the PR is required in 202405 branch, please file a direct PR to 202405 to address cherry-pick conflict.

202405 PR -  https://github.com/sonic-net/sonic-buildimage/pull/20183

**Pull request merged**